pkg/router/template/router.go

```diff
@@ -222,18 +223,61 @@ func (r *templateRouter) EnableRateLimiter(interval int, handlerFunc ratelimiter
 	glog.V(2).Infof("Template router will coalesce reloads within %v seconds of each other", interval)
 }
 
+// When there is no default cert, cobble one together from
+// the mounted secret, if present
+func secretToPem(outName string) error {
+	// The secret, when present, is mounted on /etc/pki/tls/private
+	// The secret has two components crt.tls and key.tls
+	// When the default cert is provided by the admin it is a pem
+	//   tls.crt is the supplied pem and tls.key is the key
+	//   extracted from the pem
+	// When the admin does not provide a default cert, the secret
+	//   is created via the service annotation. In this case
+	//   tls.crt is the cert and tls.key is the key
+	//   The crt and key are concatenated to form the needed pem
+	// is /etc/pki/tls/private/tls.crt and tls.key present
+	var fileCrtName = "/etc/pki/tls/private/tls.crt"
```

Comment:
OK I added the env var. We are here because the router couldn't get the default route. The last chance is to look for a secret.